core/src/main/java/com/graphhopper/routing/util/WheelchairFlagEncoder.java

```diff
@@ -111,6 +117,25 @@ public int getVersion() {
         return 1;
     }
 
+    /**
+     * Remove speedEncoder from super class FootFlagEncoder and add a speedEncoder which supports two directions for
+     * speed calculation at slopes.
+     * @param registerNewEncodedValue
+     * @param prefix
+     * @param index
+     */
+    @Override
+    public void createEncodedValues(List<EncodedValue> registerNewEncodedValue, String prefix, int index) {
+        super.createEncodedValues(registerNewEncodedValue, prefix, index);
+        for (int i = 0; i < registerNewEncodedValue.size(); i++) {
+            if (registerNewEncodedValue.get(i).getName().endsWith("average_speed")) {
+                registerNewEncodedValue.remove(i);
+                registerNewEncodedValue.add(speedEncoder = new UnsignedDecimalEncodedValue(
+                        getKey(prefix, "average_speed"), speedBits, speedFactor, true));
+            }
+        }
+    }
```

Comment:
It would be simpler to make the property configurable in FootFlagEncoder and then set storeTwoDirections=true in this encoder. See https://github.com/graphhopper/graphhopper/blob/master/core/src/main/java/com/graphhopper/routing/util/MotorcycleFlagEncoder.java#L67

Reply:
Please remove this method and introduce speedTwoDirections in the super class